pkg/envoy/lds/response_test.go

```diff
@@ -0,0 +1,99 @@
+package lds
+
+import (
+	"fmt"
+	"testing"
+
+	xds_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	xds_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
+	"github.com/golang/mock/gomock"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/google/uuid"
+	tassert "github.com/stretchr/testify/assert"
+	"k8s.io/client-go/kubernetes"
+	testclient "k8s.io/client-go/kubernetes/fake"
+
+	"github.com/openservicemesh/osm/pkg/catalog"
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/configurator"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/tests"
+)
+
+func getProxy(kubeClient kubernetes.Interface) (*envoy.Proxy, error) {
+	if _, err := tests.MakePod(kubeClient, tests.Namespace, tests.BookbuyerServiceName, tests.BookbuyerServiceAccountName, tests.ProxyUUID); err != nil {
+		return nil, err
+	}
+	if _, err := tests.MakePod(kubeClient, tests.Namespace, "bookstore", tests.BookstoreServiceAccountName, uuid.New().String()); err != nil {
+		return nil, err
+	}
+
+	for _, svcName := range []string{tests.BookbuyerServiceName, tests.BookstoreApexServiceName, tests.BookstoreV1ServiceName, tests.BookstoreV2ServiceName} {
+		if _, err := tests.MakeService(kubeClient, svcName); err != nil {
+			return nil, err
+		}
+	}
+
+	certCommonName := certificate.CommonName(fmt.Sprintf("%s.%s.%s", tests.ProxyUUID, tests.BookbuyerServiceAccountName, tests.Namespace))
+	certSerialNumber := certificate.SerialNumber("123456")
+	proxy := envoy.NewProxy(certCommonName, certSerialNumber, nil)
+	return proxy, nil
+}
+
+func TestListenerConfiguration(t *testing.T) {
+	assert := tassert.New(t)
+	mockCtrl := gomock.NewController(t)
+	mockConfigurator := configurator.NewMockConfigurator(mockCtrl)
+	kubeClient := testclient.NewSimpleClientset()
+	meshCatalog := catalog.NewFakeMeshCatalog(kubeClient)
+
+	proxy, err := getProxy(kubeClient)
+	assert.Empty(err)
+	assert.NotNil(meshCatalog)
+	assert.NotNil(proxy)
+
+	mockConfigurator.EXPECT().IsPermissiveTrafficPolicyMode().Return(false).AnyTimes()
+	mockConfigurator.EXPECT().IsPrometheusScrapingEnabled().Return(true).AnyTimes()
+	mockConfigurator.EXPECT().IsTracingEnabled().Return(false).AnyTimes()
+	mockConfigurator.EXPECT().IsEgressEnabled().Return(true).AnyTimes()
+
+	actual, err := NewResponse(meshCatalog, proxy, nil, mockConfigurator, nil)
+	assert.Empty(err)
+	assert.NotNil(actual)
+	assert.Len(actual.Resources, 3)
```

Comment:
next to hardcoded expectations such as these, could you drop a comment similar to the below, doesn't have to be as verbose.
https://github.com/openservicemesh/osm/blob/e3356c54e825428023bb8be68c9a491c461cbe71/pkg/envoy/cds/response_test.go#L87-L95